packages/back-end/src/models/ExperimentSnapshotModel.ts

```diff
@@ -255,18 +255,20 @@ export async function updateSnapshot({
       : false;
 
     if (experimentModel && isLatestPhase) {
-      await updateExperimentAnalysisSummary({
+      const updatedExperiment = await updateExperimentAnalysisSummary({
         context,
         experiment: experimentModel,
         experimentSnapshot: experimentSnapshotModel,
       });
+
+      await notifyExperimentChange({
```

Comment:
question: We are calling this for less snapshots/experiments now as we have moved it inside `if (shouldUpdateExperimentAnalysisSummary) && (experimentModel && isLatestPhase)`.

Just making sure this is intentional, as it can affect other notification and not just the new Decision Framework notifications.

potential issue: This in conjunction with the removal of the checks inside `notifyExperimentChange` means we will not be calling this function for bandit snapshots for example.

Reply:
Fewer snapshots:

Maybe? If you look at the code inside `notifyExperimentChange` it kind of does many of the exact same checks and I wanted to put more of the logic in one place instead of having summary update for some snapshots and notifications fire for others.

In truth, the current experiment status and the notifications that are behind it should probably only ever update at the same time. If there are deviations, we want to be very intentional about it.

Bandits:

Why would we no longer be calling this for bandits? Bandits have an `experimentModel` and we only want to fire for any analysis done in the `latestPhase`. They also will be *snapshot* type === "standard"?

Reply:
Thanks for the explanation!

True about bandit (I got `snapshot.type` mixed up with `experiment.type`), so please disregard.

I think these changes make sense and we should proceed!